File: Segundo/Monopoly/civitas_java (1)/proyecto_civitas/src/civitas/SorpresaPorJugador.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package civitas;

/**
 *
 * @author pc
 */
import java.util.ArrayList;
public class SorpresaPorJugador extends Sorpresa{
    private int valor;

    SorpresaPorJugador(int valor,String texto)
    {
        super(texto);
        this.valor=valor;
    }
    @Override
    public void aplicarAJugador(int actual, ArrayList<Jugador> todos){
         if(jugadorCorrecto(actual,todos))
         {
            informe(actual,todos);
            SorpresaPagarCobrar a = new SorpresaPagarCobrar(valor*-1,"Pagan los demas");
            for(int i=0;i<todos.size();i++)
            {
                if(i!=actual)
                    a.aplicarAJugador(i, todos);
            }
             
            SorpresaPagarCobrar b = new SorpresaPagarCobrar(valor*(todos.size()-1)," ,y recibe dinero de los demas");
            b.aplicarAJugador(actual, todos);
        }
    }
    @Override
    public String toString() {
        return "Sorpresa Aplicar A Jugador " + "valor: " + valor + super.toString()+ '\n';
    }
}

File: Segundo/Monopoly/civitas_ruby (1)/proyecto_civitas_ruby/lib/sorpresa_por_jugador.rb
# To change this license header, choose License Headers in Project Properties.
# To change this template file, choose Tools | Templates
# and open the template in the editor.

module Civitas
class SorpresaPorJugador < Sorpresa
  def initialize(texto,valor)
    super(texto)
    @valor = valor
  end
  
  def aplicarAJugador(actual,todos)
     if(jugadorCorrecto(actual,todos))
        super(actual,todos)
        s1=SorpresaPagarCobrar.new("Pagan todos los jugadores menos el no afectado ",@valor * -1)
        var = 0  
        while var < todos.size
          if(var!=actual)
            s1.aplicarAJugador(var,todos)
          end
          var+=1
        end
        s2=SorpresaPagarCobrar.new("Recibe dinero de todos los jugadores",@valor*(todos.size-1))
        s2.aplicarAJugador(actual,todos)
    end
  end
  
  def toString
    a = "Por Jugador"
    return a
  end
  public_class_method :new
end
end